client/faucet/src/main.rs

```diff
@@ -309,11 +341,17 @@ mod tests {
                                 .insert(address, create_vasp_account(&address.to_string(), 0));
                             assert!(previous.is_none(), "should not create account twice");
                         }
+                        Some(ScriptCall::CreateDesignatedDealer { addr: address, .. }) => {
+                            let mut writer = accounts.write();
+                            let previous =
+                                writer.insert(address, create_dd_account(&address.to_string(), 0));
+                            assert!(previous.is_none(), "should not create account twice");
+                        }
                         Some(ScriptCall::PeerToPeerWithMetadata { payee, amount, .. }) => {
                             let mut writer = accounts.write();
                             let account =
                                 writer.get_mut(&payee).expect("account should be created");
-                            *account = create_vasp_account(payee.to_string().as_str(), amount);
```

Comment:
How come this is not needed anymore?

Reply:
new implementation direct update the balances amount:  account["balances"][0]["amount"] = serde_json::json!(amount);

old implementation recreates vasp account json, but won't work well anymore because depending on payee account role, it maybe DD or VASP.